Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/BackgroundNoiseOneTimeWorker.kt

```diff
@@ -41,4 +40,7 @@ class BackgroundNoiseOneTimeWorker(
 
         return result
     }
+
+    @AssistedInject.Factory
+    interface Factory : InjectedWorkerFactory<BackgroundNoiseOneTimeWorker>
```

Comment:
Every worker needs a factory. The `InjectedWorkerFactory` just hides the boilerplate code:

```
interface Factory {
    fun create(context: Context, workerParams: WorkerParameters)
}
```

Reply:
Note: There is one peculiarity with `AssistedInject`, the parameter names of the `@AssistedInject.Factory` need to be written exactly the same as the constructor arguments annotated with `@Assisted`.